### PR TITLE
Added docstring and refactor metadata to `IntanRecordingInterface`

### PR DESCRIPTION
**Pull request by h-mayorquin**
Added docstring to intan and some refactoring of the`get_metadat` function. This should be done after #561.

### CODE REVIEW

File: src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py [hunk 1]
```diff
@@ -79,6 +79,21 @@ def __init__(
         spikeextractors_backend: Optional[bool] = False,
```
Comment (bendichter) [inline src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py:79]:
```suggestion
        spikeextractors_backend: bool = False,
```

File: src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py [hunk 2]
```diff
@@ -79,6 +79,21 @@ def __init__(
...
+        """Load and prepare raw data and corresponding metadata from the Intan format (.rhd or .rhs files).
+
+
+        Parameters
+        ----------
+        file_path : FilePathType
+            Path to either a rhd or a rhs file
+        stream_id : Optional[str], optional
+            The stream of the data for spikeinterface, "0" by default.
+        spikeextractors_backend : Optional[bool], optional
```
Comment (bendichter) [inline src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py:91]:
```suggestion
        spikeextractors_backend : bool
```

File: src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py [hunk 3]
```diff
@@ -79,6 +79,21 @@ def __init__(
...
+            False by default. When True the interface uses the old extractor from the spikextractors library instead
+            of a new spikeinterface object.
+        verbose : Optional[bool], optional
```
Comment (bendichter) [inline src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py:94]:
```suggestion
        spikeextractors_backend: bool = False,
        verbose: bool = True,
    ):
        """Load and prepare raw data and corresponding metadata from the Intan format (.rhd or .rhs files).


        Parameters
        ----------
        file_path : FilePathType
            Path to either a rhd or a rhs file
        stream_id : str, optional
            The stream of the data for spikeinterface, "0" by default.
        spikeextractors_backend : bool
            False by default. When True the interface uses the old extractor from the spikextractors library instead
            of a new spikeinterface object.
        verbose : bool
```

Comment (bendichter) [reply src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py:94]: `Optional` is used from accepting the `None` value, so it is not appropriate here: https://docs.python.org/3/library/typing.html#typing.Optional

Comment (h-mayorquin) [reply src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py]: Thanks, I added this somewhere else where it should not be. I will fix.

File: src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py [hunk 4]
```diff
@@ -114,43 +129,50 @@ def get_metadata_schema(self):
...
             )
         if "custom_channel_name" in recording_extractor_properties:
-            ecephys_metadata["Ecephys"]["Electrodes"].append(
+            ecephys_metadata["Electrodes"].append(
                 dict(name="custom_channel_name", description="Custom channel name assigned in Intan.")
             )
 
-        return ecephys_metadata
+        # Build metadata dictionary and return complete structure
+        metadata = dict()
```
Comment (bendichter) [inline src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py:176]: you aren't calling `super().get_metadata`?

Comment (h-mayorquin) [reply src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py:176]: I just went through the git history of the file and it was never there. Let me check if this allows for code reduction.

Comment (h-mayorquin) [reply src/nwb_conversion_tools/datainterfaces/ecephys/intan/intandatainterface.py]: Added in in the last commit.

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin So this just needs metadata tests now for the Intan interface, right?

**Comment by h-mayorquin** [comment]
> @h-mayorquin So this just needs metadata tests now for the Intan interface, right?

Yes. Along with most of the interfaces.

**Pull request merged**